hoot-core/src/main/cpp/hoot/core/conflate/highway/HighwayMatch.cpp

```diff
@@ -73,13 +77,31 @@ HighwayMatch::HighwayMatch(const shared_ptr<HighwayClassifier>& classifier,
     {
       // calculate the match score
       _c = _classifier->classify(map, eid1, eid2, _sublineMatch);
-      stringstream ss;
-      ss << mt->getTypeDetail(_c);
-      _explainText = QString::fromUtf8(ss.str().data());
 
-      stringstream ss2;
-      ss2 << toString() << " features: " << getFeatures(map);
-      //LOG_DEBUG(QString::fromUtf8(ss2.str().data()));
+      MatchType type = getType();
+      QStringList description;
+      if (type != MatchType::Match)
+      {
+        //  Check the Angle Histogram
+        double angle = AngleHistogramExtractor().extract(*map, e1, e2);
+        if (angle >= 0.75)          description.append("Very similar highway orientation.");
+        else if (angle >= 0.5)      description.append("Similar highway orientation.");
+        else if (angle >= 0.25)     description.append("Semi-similar highway orientation.");
+        else                        description.append("Highway orientation not similar.");
+        //  Use the average of the edge distance extractors
+        double edge = (EdgeDistanceExtractor(new RmseAggregator()).extract(*map, e1, e2) +
+                       EdgeDistanceExtractor(new SigmaAggregator()).extract(*map, e1, e2)) / 2.0;
+        if (edge >= 90)             description.append("Highway edges very close to each other.");
+        else if (edge >= 70)        description.append("Highway edges somewhat close to each other.");
+        else                        description.append("Highway edges not very close to each other.");
+      }
+
+      if (description.length() > 0)
+        _explainText = description.join(" ");
+      else
+        _explainText = mt->getTypeDetail(_c);
+      //  WTF?! This has to run or the whole thing gets screwed up?!
+      getFeatures(map);
```

Comment:
@jasonsurratt @mattjdnv This is an interesting issue that I've come across, when I added in the new review explanation text I initially removed the following code all together because `ss2` wasn't being used and a bunch of tests failed.
```
      stringstream ss2;
      ss2 << toString() << " features: " << getFeatures(map);
      //LOG_DEBUG(QString::fromUtf8(ss2.str().data()));
```
I found that by calling `getFeatures(map)` the tests all work again as they should.  I took the first test that failed `test-files/cmd/slow/ConflateCmdHighwayExactMatchInputsTest.sh` and compared the output with the expected output (see the attached screenshot).
![getfeatures](https://cloud.githubusercontent.com/assets/13385275/16157655/0edfe8e2-347f-11e6-9be5-5d504a6b084f.png)
From what I'm seeing `getFeatures(map)` changes the way ids.  I don't know if this is desirable or not.